apps/glusterfs/app_block_volume_test.go

```diff
@@ -492,6 +499,51 @@ func TestBlockVolumeCreateHACountHostUnavailableFail(t *testing.T) {
 	tests.Assert(t, countTrue(hostAvailable) == 2)
 }
 
+func TestBlockVolumeCreateHAShuffled(t *testing.T) {
+	tmpfile := tests.Tempfile()
+	defer os.Remove(tmpfile)
+
+	// Create the app
+	app := NewTestApp(tmpfile)
+	defer app.Close()
+	router := mux.NewRouter()
+	app.SetRoutes(router)
+
+	// Setup the server
+	ts := httptest.NewServer(router)
+	defer ts.Close()
+
+	// Setup database
+	err := setupSampleDbWithTopology(app,
+		1,    // clusters
+		10,   // nodes_per_cluster
+		10,   // devices_per_node,
+		5*TB, // disksize)
+	)
+	tests.Assert(t, err == nil)
+
+	// BlockVolumeCreate
+	request := []byte(`{"size": 100, "hacount": 3}`)
+
+	var prevHosts []string
+	for i := 0; i < 5; i++ {
+		r, err := http.Post(ts.URL+"/blockvolumes", "application/json", bytes.NewBuffer(request))
+		tests.Assert(t, err == nil)
+		info := blockVolumeTestResult(t, r)
+
+		// check that the hosts counts are as expected and match the ha counts
+		// verify that we do not get the same hosts every time this function is
+		// called.
+		tests.Assert(t, len(info.BlockVolume.Hosts) == 3)
+		tests.Assert(t, info.Hacount == 3)
+		tests.Assert(t, !reflect.DeepEqual(prevHosts, info.BlockVolume.Hosts),
```

Comment:
I expect this test to fail every 3.6 millionth run on average, because that is the likelihood, that shuffling of 10 elements returns the identical sequence. That's unlikely... I know... but... it would still itch me... ;-)